libsql-server/src/namespace/mod.rs

```diff
@@ -495,6 +497,15 @@ impl<M: MakeNamespace> NamespaceStore<M> {
         Ok(())
     }
 
+    pub async fn shutdown(&self) -> crate::Result<()> {
```

Comment:
you should mark the store as shut down to prevent other namespaces to be reallocated after this is called

Reply:
Could you elaborate? Do you simply mean keep some variable like `has_shutdown`, set it to true at shutdown and use this boolean at other places while creating/forking namespace?

Reply:
you won't be able to take ownership because you only have a ref, so yes, the first option sounds great :)